build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/GenerateUnixFilePermissions.cs

```diff
@@ -0,0 +1,52 @@
+using System.IO;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	/// <summary>
+	/// Creates data/UnixFilePermissions.xml
+	/// </summary>
+	public class GenerateUnixFilePermissions : Task
+	{
+		[Required]
+		public string Output { get; set; }
+
+		public string PackagePath { get; set; } = "";
+
+		public ITaskItem [] Files { get; set; }
+
+		public override bool Execute ()
+		{
+			var settings = new XmlWriterSettings {
+				OmitXmlDeclaration = true,
+				Indent = true,
+			};
+
+			/*
+			 <FileList>
+			   <File Path="tools/Darwin/aapt2" Permission="755" />
+			 </FileList>
+			*/
+
+			using var xml = XmlWriter.Create (Output, settings);
+			xml.WriteStartElement ("FileList");
+			if (Files != null) {
+				foreach (var file in Files) {
+					var path = file.GetMetadata ("RelativePath");
+					var permission = file.GetMetadata ("Permission");
+					if (!string.IsNullOrEmpty (path) && !string.IsNullOrEmpty (permission)) {
```

Comment:
Please "invert" this:

```csharp
if (string.IsNullOrEmpty (path) || string.IsNullOrEmpty (permission)) {
    continue;
}
xml.WriteStartElement (…)
…
```

Reply:
The latest uses LINQ to `order by`, so we lost that indentation.